### ISSUE TITLE
Experiment result wrongly produced for CIFAR10, it's not even spikes that you're training the model with

### ISSUE THREAD

**Issue by manh-vnguyen** [Intelli-Chip-Lab/snn_temporal_decoupling_distillation#3]
experiment/cifar/main.py
Line 41-61

```
ptns, labels = ptns.to(device), labels.to(device)
if model.step_mode == 's':
            out_spikes = []
            for t in range(args.T):
                out = model(ptns)
                out_spikes.append(out)
            output = torch.stack(out_spikes, dim=0)
            avg_fr = output.mean(dim=0)
        else:
            in_data, _ = torch.broadcast_tensors(ptns, torch.zeros((args.T,) + ptns.shape))
            in_data = in_data.reshape(-1, *in_data.shape[2:])
            output = model(in_data)
            with torch.no_grad():
                t_avg_fr = t_model(ptns)
            avg_fr =output.mean(dim=0)

```

Essentially, no spike train is generated from the image tensor, the float value is passing directly into the model, which is more information. The problem with spiking neural network is the lesser of accuracy in feeding data with spike train, thus in not doing this you essentially defeat the purpose of improving the accuracy of spiking neural network!

**Comment by manh-vnguyen** [comment]
After doing literature review and looking at the available codes of other papers. It seems like they are also forwarding input into the model in similar manner. This modeling doesn't appear in a survey paper by J. Eshragian, but seems like other researchers are also using it. I'm still confused as to where the basis of this comes from but I guess it's not on the author to explain this for their paper. So this is not really an issue.
